native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -202,8 +202,9 @@ private static void processAttributeWithAnnotation(BXmlItem xmlItem, String attr
                 } else if (fieldType instanceof UnionType) {
                     UnionType bUnionType = (UnionType) fieldType;
                     for (Type memberType : bUnionType.getMemberTypes()) {
-                        if (memberType.getTag() == TypeTags.RECORD_TYPE_TAG)  {
-                            annotationType = memberType;
+                        Type referMemberType = TypeUtils.getReferredType(memberType);
```

Comment:
Line 200-201 not required if the reference type is resolved as soon as the field type is accessed in line 197.

@kalaiyarasiganeshalingam Verify whether there are other usages like this and update there. This can lead to regression issues if not looked into properly

Reply:
Updated in [f8e2390](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/465/commits/f8e2390d80c78b7b837eebdb5a831547e4d51f4b)

Verified other places as well.